### PR TITLE
pull all playlists instead of just first page

### LINKED ISSUE #2: Pull all playlists instead of just the first page
The max limit per page of playlists is 50. We need to pull all playlists.

API reference for offset: https://developer.spotify.com/web-api/get-list-users-playlists/

Logic is in `getPlaylists()` method of `spotify-service.js`.

Going to need some kind of Promise API fancy footwork to continually pull and update offset.

Call to get user playlists returns a total, we can simply divide the limit by the offset to know how many times to pull. There's also a `next` property; we can check for that being null to know when we're done. 

I think I prefer using offset / limit to get the number of times to pull.

### PR DESCRIPTION
**Pull request by alexanderpepper**
Fixes #2

### CODE REVIEW

File: server/services/spotify-service.js [hunk 1]
```diff
@@ -38,11 +38,12 @@ module.exports = class SpotifyService {
...
-    // this.setCredentials()
-    return spotifyApi.getMe().then(data => {
-      console.log('getMe() returned', data.body)
-      return spotifyApi.getUserPlaylists(data.body.id, {limit: 50}).then(data => data.body)
-    })
+  async getPlaylists () {
+    const user = await spotifyApi.getMe().then(data => data.body)
+    const sampling = await spotifyApi.getUserPlaylists(user.id, {limit: 1}).then(data => data.body)
+    const pageCount = Math.ceil(sampling.total / limit)
+    const pages = Array.from(Array(pageCount), (_, x) => x)
```
Comment (macacollins) [inline server/services/spotify-service.js]: Consider `Array.from(Array(pageCount).keys())`. The mapFn argument here tripped me up a bit until I read more.

File: server/services/spotify-service.js [hunk 2]
```diff
@@ -38,11 +38,12 @@ module.exports = class SpotifyService {
...
+    const promises = pages.map(p => spotifyApi.getUserPlaylists(user.id, {limit, offset: p * limit}).then(data => data.body.items))
+    return Promise.all(promises).then(results => results.reduce((acc, val) => [...acc, ...val]))
```
Comment (macacollins) [inline server/services/spotify-service.js]: Isn't `[...acc, ...val]` here basically `acc.concat(val)`? Or am I missing something?

File: server/services/spotify-service.js (hunk 1 above)
Comment (alexanderpepper) [reply server/services/spotify-service.js]: Yeah .keys() is much more readable. Will change that.

File: server/services/spotify-service.js (hunk 2 above)
Comment (alexanderpepper) [reply server/services/spotify-service.js]: Yes it is. I'll make that switch. Spread operator was being too fancy here.

**Comment by macacollins** [comment]
👍

**Pull request merged**